src/components/ProductTours/AdminOnboardingTours/data/useAdminOnboardingTour.tsx

```diff
@@ -0,0 +1,67 @@
+import { sendEnterpriseTrackEvent } from '@edx/frontend-enterprise-utils';
+import { logError } from '@edx/frontend-platform/logging';
+
+import {
+  ANALYTICS_INSIGHTS_TARGETS,
+  ORGANIZE_LEARNER_TARGETS,
+  TRACK_LEARNER_PROGRESS_TARGETS,
+} from '../constants';
+
+import LmsApiService from '../../../../data/services/LmsApiService';
+import useCreateLearnerProgressFlow from './useCreateLearnerProgressFlow';
+import useCreateAnalyticsFlow from './useCreateAnalyticsFlow';
+import { TourStep } from '../../types';
+import useCreateOrganizeLearnersFlow from './useCreateOrganizeLearnersFlow';
+
+interface UseAdminOnboardingTourProps {
+  adminUuid: string;
+  aiButtonVisible: boolean;
+  currentStep: number;
+  setCurrentStep: (currentStep: number) => void;
+  enterpriseSlug: string;
+  targetSelector?: string;
+}
+
+const useAdminOnboardingTour = (
+  {
+    adminUuid, aiButtonVisible, currentStep, setCurrentStep, enterpriseSlug, targetSelector,
+  }: UseAdminOnboardingTourProps,
+): Array<TourStep> => {
+  const handleAdvanceTour = (advanceEventName: string) => {
+    const newIndex = currentStep + 1;
+    sendEnterpriseTrackEvent(enterpriseSlug, advanceEventName, { 'completed-step': newIndex });
+    setCurrentStep(newIndex);
```

Comment:
Since this is business logic operating with data, I would say we should unit test these functions.
Treating them as hooks also makes them difficult to unit test, but turning them into normal functions allows you to just call them directly in a unit test and just check that they are doing everything correctly.